Add paging toolbar to search results and update results text

adamtsai1 opened the pull request:
@kellypacker I didn't focus a ton on styling since I imagine we'll give it a little design love if we ever launch this, but let me know if anything jumps out at you.

kellypacker commented:
@adamtsai1 this looks good. I think the pagination should be hidden if there is just one page. The Last/First buttons seems like they add visual noise, but might not be used that much, even if there are more pages. I would think the filters would be used to narrow things down instead. What do you think?

adamtsai1 commented:
@kellypacker I've added logic to hide the pagination if it's only 1 page, and you're probably right about the first/last so I've removed those. Since these are simple changes, I went ahead and merged to master.
